src/OptionSettlement.sol

```diff
@@ -188,31 +188,52 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
 
     /// @inheritdoc IOptionSettlementEngine
     function write(uint256 optionId, uint112 amount) external returns (uint256 claimId) {
-        if (amount == 0) {
-            revert AmountWrittenCannotBeZero();
-        }
+        /// claimId == 0 signifies that a new claim NFT should be minted
+        return write(0, optionId, amount);
+    }
 
+    /// @inheritdoc IOptionSettlementEngine
+    function write(uint256 claimId, uint256 optionId, uint112 amount) public returns (uint256) {
         (uint160 _optionId, uint96 claimIndex) = getDecodedIdComponents(optionId);
+
+        // claim index must be zero in lower 96b for provided option Id
```

Comment:
Why? I don't think this is true.

Reply:
the 'why' is bc it wouldn't make sense to call write(<claim id>, <amount>) the first argument of write() should always be an option ID. option IDs, from our encoding scheme, necessarily have the lower 96b set to zero. The claimIndex referred to here is just those lower 96b.

Reply:
an improvement here would be to retitle the claimIndex var to make this clearer